build/scripts/build_index_image.sh

```diff
@@ -110,6 +110,9 @@ if [ "$RELEASE" == "true" ]; then
   # shellcheck disable=SC2016
   yq -Yi --arg replaces "$LATEST_BUGFIX" '.spec.replaces = $replaces' \
     deploy/templates/components/csv/clusterserviceversion.yaml
+
+  # Set digests instead of tags, it allows to mirror images in disconnected environment
+  . build/scripts/set_digests.sh -c deploy/templates/components/csv/clusterserviceversion.yaml
```

Comment:
There is still an issue here, as `build_index_image.sh` is called _before_ building the controller and project clone images, so we'll still end up with empty digests for those :/

There's a fundamental issue here that I'm struggling to work around (see [this branch](https://github.com/amisevsk/devworkspace-operator/tree/use-digests-in-olm-release-wip) for today's efforts). 

* The index is built based off the files in `./olm-catalog`, which need to contain the digest of the controller image (i.e. have to build controller image before committing)
* The controller image refers to the commit its built from in its version information (i.e. have to build controller image after committing)

I'm not sure using digests in the catalog is possible to do with only one repo, if we're using file-based catalogs. 

I'm trying to work out a way to do this as we intend, but it's going to be complicated.

Reply:
I found two more issues here:
* References to latest [1] and nightly [2] tags are not good at all.
* Replacing tags on digests in `spec.relatedImages` section is not enough.  Container image and images in evn vas must be replaced as well.

[1] `quay.io/wto/web-terminal-tooling:latest` 
[2] `quay.io/eclipse/che-machine-exec:nightly`

I can suggest creating a dedicated issue and describe all problems we have.

Reply:
Yes, I think a dedicated issue is warranted, this is more complicated than it appeared at first.

Reply:
For those images, at least, they can just be dropped: https://github.com/devfile/devworkspace-operator/pull/1039

They were needed earlier on but the functionality to use those images has since been dropped.